[Chee Zhong Wei] iP

czhongwei opened the pull request:
### DukeBot

> "With great power, comes great responsibility" - Benjamin Parker 💀

**DukeBot** revolutionises your way of remembering tasks, some features are:

- _Create_ a new Task
- _Delete_ a Task
- _List_ all Tasks
- _Mark/unmark_ a task done/incomplete
- Stores tasks in a data file

To get started:
1. Go to this [link](https://github.com/czhongwei/ip)
2. Navigate to release
3. Download the .jar file
4. Run it

You can:
- [ ] Share
- [ ] Watch
- [ ] Star
the repository

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Duke {
    public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }
}
```

Commit:
Add Gradle support

czhongwei committed:
Added echoing to Thesh

czhongwei committed:
Added tasks array and reworked logic for handling commands

czhongwei committed:
Changed variable task_count to taskCount

czhongwei committed:
Added Task Class

czhongwei committed:
Added toString and split toggleComplete function to markComplete and markIncomplete in Task.java

czhongwei committed:
Added @Override to toString in Task.java

czhongwei committed:
Added Task object to Duke.java and ability to mark complete

czhongwei committed:
Change toString of Task.java to include Status and make relevant changes to Duke.java

czhongwei committed:
Add classes Deadline, Event, ToDo which are subclasses of Task

czhongwei committed:
Make Task class abstract

czhongwei committed:
Separate Task into Deadline, Event and ToDo

czhongwei committed:
Add indentation and lines to chatbot's dialogue

czhongwei committed:
Add EXPECTED.TXT and input.txt files for Automated Text UI Testing

czhongwei committed:
Create amd include MarkToggleException in the Task.java

czhongwei committed:
Add some Exception handling

czhongwei committed:
Add command to delete tasks

czhongwei committed:
Add Error handling for blank ToDo

czhongwei committed:
Remove unmappable character and update EXPECTED.TXT

czhongwei committed:
Remove unmappable character from Duke.java

czhongwei committed:
Add Javadocs to Task.java Deadline.java ToDo.java Event.java

czhongwei committed:
Merge pull request #1 from czhongwei/dev

Dev

czhongwei committed:
Add taskCount and methods to increase/decrease taskCount in Task.java

czhongwei committed:
Abstract mark,unmark,delete,list command of Duke

czhongwei committed:
Create DukeBot package and insert *.java files into the package

czhongwei committed:
Replace MarkToggleException with a more general DukeException

czhongwei committed:
Add getTaskType and getTime methods to Task subclasses

czhongwei committed:
Add feature to read and write from txt file to load tasks

czhongwei committed:
Change attribute of time in Deadline and Event to LocalDate type and accept datetime for timing

czhongwei committed:
Resolve merge conflicts

czhongwei committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-Gradle

czhongwei committed:
Change mainClassName in application property of build.gradle to DukeBot.Duke

czhongwei committed:
Add TaskList class and implement it in Duke.java

czhongwei committed:
Add Storage class and implement it in Duke.java

czhongwei committed:
Add Command class and its subclasses and Parser class

czhongwei committed:
Amend Duke.java to use Parser and Command classes

czhongwei committed:
Add *.txt to .gitignore

czhongwei committed:
Remove redundant code from Duke.java

czhongwei committed:
Add Ui class and implement it in Duke, TaskList and Command classes

czhongwei committed:
Move output messages of creating new tasks into Ui class

czhongwei committed:
Add testcases for EventTest and TodoTest

czhongwei committed:
Add JavaDoc to most of the non-private class/methods

czhongwei committed:
Tweak all code to comply with Java coding standard

czhongwei committed:
Add FindCommand class and implement it in the Duke program

czhongwei committed:
Fix merge conflict in Duke.java

czhongwei committed:
Fix merge conflict in Parser.java

czhongwei committed:
Add checkstyle

czhongwei committed:
Add checkstyle in build.gradle

czhongwei committed:
Remove lines derived due to merge conflicts

czhongwei committed:
Fix checkstyle violations

janelleljt commented on src/main/java/DukeBot/Ui.java:
```diff
@@ -0,0 +1,96 @@
+     * @param newTask to be printed on the output.
+     */
+    public void showNewTask(Task newTask) {
+        System.out.println("    Got it. I've added this task:");
```
```suggestion
        System.out.println("\tGot it. I've added this task:");
```
Could use `\t` to format tab in front of the phrase you are printing.

janelleljt commented on src/main/java/DukeBot/Ui.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Prints a line onto the output.
+     */
+    public void showLine() {
```
Instead of `showLine`, do you think that renaming to `printLine` might be better naming?

janelleljt commented on src/main/java/DukeBot/command/NewTaskCommand.java:
```diff
@@ -0,0 +1,49 @@
+public class NewTaskCommand extends Command {
+
+    private String command;
+
```
Can omit this empty line.

janelleljt commented on src/main/java/DukeBot/Duke.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Encapsulates the Duke class.
+ */
+public class Duke {
```
Having renamed your bot to Thesh, would it be better to rename your classes, `Duke` and `DukeException` to `Thesh` and `TheshException` as well for sake of clarity?

janelleljt reviewed (commented):
No glaring coding standard issues! I felt that your methods were well named!

czhongwei committed:
Add JavaFX files to project

czhongwei committed:
Amend .java files to accommodate JavaFX application

czhongwei committed:
Boolean isExit attribute of Duke is public

Access modifier of boolean isExit of Duke.java is non-private which does not conform with OOP

Change access modifier of boolean isExit in Duke.java to private
Change duke.isExit call in MainWindow.java to duke.isExit()

czhongwei committed:
Add assert statements in Duke.java, Storage.java and MainWindow.java

czhongwei committed:
Change some files to adhere to Code Quality

czhongwei committed:
Merge pull request #2 from czhongwei/branch-A-Assertions

Add Assert statements to some files

czhongwei committed:
Merge branch 'master' into branch-A-CodeQuality

czhongwei committed:
Merge pull request #3 from czhongwei/branch-A-CodeQuality

Amend codebase to adhere to code quality

czhongwei committed:
Add Priority Enumeration

czhongwei committed:
Add support for priority and change in priority commands

czhongwei committed:
Add support for storing and loading tasks to and from a file

czhongwei committed:
Merge pull request #5 from czhongwei/branch-C-Priority

Implement C-Priority

czhongwei committed:
Fix coding standard violations

czhongwei committed:
Print additional blank line in showNewTask method

czhongwei committed:
Amend exception message of create deadline command

czhongwei committed:
Adjust size of DialogBox and GUI

czhongwei committed:
Change padding of DialogBox

czhongwei committed:
Add image of how the user interface looks like in docs folder

czhongwei committed:
Add blank line in unmark command response

czhongwei committed:
Edit README.md to add the UG

czhongwei committed:
Change path of storage file

czhongwei committed:
Change mainClassName and archiveBaseName in build.gradle
